server/src/test/java/org/elasticsearch/gateway/IncrementalClusterStateWriterTests.java

```diff
@@ -173,56 +154,44 @@ private IndexMetaData createIndexMetaData(String name) {
     public void testGetRelevantIndicesWithUnassignedShardsOnMasterEligibleNode() {
         IndexMetaData indexMetaData = createIndexMetaData("test");
         Set<Index> indices = IncrementalClusterStateWriter.getRelevantIndices(
-            clusterStateWithUnassignedIndex(indexMetaData, true),
-            noIndexClusterState(true),
-            randomPrevWrittenIndices(indexMetaData));
+            clusterStateWithUnassignedIndex(indexMetaData, true));
         assertThat(indices.size(), equalTo(1));
     }
 
     public void testGetRelevantIndicesWithUnassignedShardsOnDataOnlyNode() {
         IndexMetaData indexMetaData = createIndexMetaData("test");
         Set<Index> indices = IncrementalClusterStateWriter.getRelevantIndices(
-            clusterStateWithUnassignedIndex(indexMetaData, false),
-            noIndexClusterState(false),
-            randomPrevWrittenIndices(indexMetaData));
+            clusterStateWithUnassignedIndex(indexMetaData, false));
         assertThat(indices.size(), equalTo(0));
     }
 
     public void testGetRelevantIndicesWithAssignedShards() {
         IndexMetaData indexMetaData = createIndexMetaData("test");
         boolean masterEligible = randomBoolean();
         Set<Index> indices = IncrementalClusterStateWriter.getRelevantIndices(
-            clusterStateWithAssignedIndex(indexMetaData, masterEligible),
-            clusterStateWithUnassignedIndex(indexMetaData, masterEligible),
-            randomPrevWrittenIndices(indexMetaData));
+            clusterStateWithAssignedIndex(indexMetaData, masterEligible));
         assertThat(indices.size(), equalTo(1));
     }
 
     public void testGetRelevantIndicesForClosedPrevWrittenIndexOnDataOnlyNode() {
         IndexMetaData indexMetaData = createIndexMetaData("test");
         Set<Index> indices = IncrementalClusterStateWriter.getRelevantIndices(
-            clusterStateWithClosedIndex(indexMetaData, false),
-            clusterStateWithAssignedIndex(indexMetaData, false),
-            Collections.singleton(indexMetaData.getIndex()));
-        assertThat(indices.size(), equalTo(1));
+            clusterStateWithClosedIndex(indexMetaData, false));
+        assertThat(indices.size(), equalTo(0));
     }
 
     public void testGetRelevantIndicesForClosedPrevNotWrittenIndexOnDataOnlyNode() {
         IndexMetaData indexMetaData = createIndexMetaData("test");
         Set<Index> indices = IncrementalClusterStateWriter.getRelevantIndices(
-            clusterStateWithJustOpenedIndex(indexMetaData, false),
-            clusterStateWithClosedIndex(indexMetaData, false),
-            Collections.emptySet());
+            clusterStateWithJustOpenedIndex(indexMetaData, false));
         assertThat(indices.size(), equalTo(0));
     }
 
     public void testGetRelevantIndicesForWasClosedPrevWrittenIndexOnDataOnlyNode() {
         IndexMetaData indexMetaData = createIndexMetaData("test");
```

Comment:
This test is now a duplicate of the previous one, and I think that `PrevNotWritten/PrevWritten` becomes misleading now `getRelevantIndicesOnDataOnlyNode` is changed. I'd prefer to have `testGetRelevantIndicesForClosedIndexOnDataOnlyNode` testing a closed index replicated/not yet replicated

Reply:
makes sense,  updated